diff -burN binutils-2.19.1.org/bfd/config.bfd binutils-2.19.1/bfd/config.bfd
--- binutils-2.19.1.org/bfd/config.bfd	2009-05-18 15:18:05.000000000 +0200
+++ binutils-2.19.1/bfd/config.bfd	2009-05-18 15:27:05.000000000 +0200
@@ -531,6 +531,11 @@
     targ_selvecs="mach_o_le_vec mach_o_be_vec mach_o_fat_vec pef_vec pef_xlib_vec sym_vec"
     targ_archs="bfd_i386_arch bfd_powerpc_arch bfd_rs6000_arch"
     ;;
+  i[3-7]86-*-iPosix*)
+  	targ_defvec=bfd_elf32_i386_vec
+	targ_selvecs=i386coff_vec
+	targ64_selvecs=bfd_elf64_x86_64_vec
+	;;
  i[3-7]86-sequent-bsd*)
     targ_defvec=i386dynix_vec
     targ_underscore=yes
diff -burN binutils-2.19.1.org/config.sub binutils-2.19.1/config.sub
--- binutils-2.19.1.org/config.sub	2009-05-18 15:18:17.000000000 +0200
+++ binutils-2.19.1/config.sub	2009-05-18 15:04:23.000000000 +0200
@@ -1247,7 +1247,7 @@
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -solaris* | -sym* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* \
+	      | -aos* | -iPosix* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff -burN binutils-2.19.1.org/configure binutils-2.19.1/configure
--- binutils-2.19.1.org/configure	2009-05-18 15:18:17.000000000 +0200
+++ binutils-2.19.1/configure	2009-05-18 15:40:34.000000000 +0200
@@ -1881,7 +1881,7 @@
 # binutils, gas and ld appear in that order because it makes sense to run
 # "make check" in that particular order.
 # If --enable-gold is used, "gold" will replace "ld".
-host_tools="texinfo byacc flex bison binutils gas ld fixincludes gcc sid sim gdb make patch prms send-pr gprof etc expect dejagnu ash bash bzip2 m4 autoconf automake libtool diff rcs fileutils shellutils time textutils wdiff find uudecode hello tar gzip indent recode release sed utils guile perl gawk findutils gettext zip fastjar gnattools"
+host_tools="byacc flex bison binutils gas ld fixincludes gcc sid sim gdb make patch prms send-pr gprof etc expect dejagnu ash bash bzip2 m4 autoconf automake libtool diff rcs fileutils shellutils time textutils wdiff find uudecode hello tar gzip indent recode release sed utils guile perl gawk findutils gettext zip fastjar gnattools"
 
 # libgcj represents the runtime libraries only used by gcj.
 libgcj="target-libffi \
@@ -2225,6 +2225,9 @@
 	;;
     esac
     ;;
+  *-*-iPosix*)
+    noconfigdirs="$noconfigdirs ${libgcj}"
+    ;;
   *-*-kaos*)
     # Remove unsupported stuff on all kaOS configurations.
     skipdirs="target-libiberty ${libgcj} target-libstdc++-v3 target-librx"
diff -burN binutils-2.19.1.org/configure.ac binutils-2.19.1/configure.ac
--- binutils-2.19.1.org/configure.ac	2009-05-18 15:18:17.000000000 +0200
+++ binutils-2.19.1/configure.ac	2009-05-18 15:41:30.000000000 +0200
@@ -474,6 +474,9 @@
 	;;
     esac
     ;;
+  *-*-iPosix*)
+    noconfigdirs="$noconfigdirs ${libgcj}"
+    ;;
   *-*-kaos*)
     # Remove unsupported stuff on all kaOS configurations.
     skipdirs="target-libiberty ${libgcj} target-libstdc++-v3 target-librx"
diff -burN binutils-2.19.1.org/gas/configure.tgt binutils-2.19.1/gas/configure.tgt
--- binutils-2.19.1.org/gas/configure.tgt	2009-05-18 15:18:16.000000000 +0200
+++ binutils-2.19.1/gas/configure.tgt	2009-05-18 15:28:20.000000000 +0200
@@ -220,6 +220,7 @@
   i386-*-pe)				fmt=coff em=pe ;;
   i386-*-cygwin*)			fmt=coff em=pe ;;
   i386-*-interix*)			fmt=coff em=interix ;;
+  i386-*-iPosix*)			fmt=elf ;;
   i386-*-mingw*)
     case ${cpu} in
       x86_64*)				fmt=coff em=pep ;;
diff -burN binutils-2.19.1.org/ld/configure.tgt binutils-2.19.1/ld/configure.tgt
--- binutils-2.19.1.org/ld/configure.tgt	2009-05-18 15:18:02.000000000 +0200
+++ binutils-2.19.1/ld/configure.tgt	2009-05-18 15:36:07.000000000 +0200
@@ -187,6 +187,7 @@
 i[3-7]86-*-bsd386)	targ_emul=i386bsd ;;
 i[3-7]86-*-bsdi*)	targ_emul=i386bsd ;;
 i[3-7]86-*-aout)	targ_emul=i386aout ;;
+i[3-7]86-*-iPosix*) targ_emul=elf_i386 ;;
 i[3-7]86-*-linux*aout*)	targ_emul=i386linux
 			targ_extra_emuls=elf_i386
 			tdir_elf_i386=`echo ${targ_alias} | sed -e 's/aout//'` ;;
